src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -955,6 +949,36 @@ class module_sram : public module_impl
   // buffer sync to device.
   bool m_dirty{ false };
 
+  union debug_flag_union {
+    struct debug_mode_struct {
+      uint32_t dump_control_codes     : 1;
+      uint32_t dump_control_packet    : 1;
+      uint32_t dump_preemption_codes  : 1;
+      uint32_t reserved : 29;
+    } debug_flags;
+    uint32_t all;
+  }m_debug_mode = {};
+  static uint32_t s_id; //TODO: it needs come from the elf file
```

Comment:
Don't you need this as a regular non static data member.   The value is associated with the object (module_sram) once created.   If you use the static it is not specific to this object hence all the dump file names could be wrong.

```suggestion
  uint32_t m_id {0}; //TODO: it needs come from the elf file
```